update program-state-management and serialize-instruction-data-frontend (#369)
solana-foundation/developer-content: content/courses/native-onchain-development/program-state-management.md

You can test your program by submitting a transaction with the right instruction
data. For that, feel free to use
[this script](https://github.com/Unboxed-Software/solana-movie-client) or

> [the frontend](https://github.com/EmekaManuel/movie-review-dapp) we built in the

Reviewer: Thanks for your work here. I've cloned the repo to the `solana-developers` organization (where Solana Devrel does it's work). 

```suggestion
[the frontend](https://github.com/solana-developers/movie-review-dapp) we built in the
```

Author: > Thanks for your work here. I've cloned the repo (including all branches) to `movie-review-frontend` under the `solana-developers` organization (where Solana Devrel does it's work) and invited you as a collaborator.
> 
> Please update all the URLs to match!

updated all URLs to match

[Deserialize Custom Instruction Data lesson](/content/courses/native-onchain-development/deserialize-custom-data-frontend.md).
In both cases, make sure you copy and paste the program ID for your program into

> the appropriate area of the source code to make sure you're testing the right

Reviewer: Name the specific file.

Reviewer: PS. Looking elsewhere, if the program ID is in two places, make it be in one place rather than duplicate the code.

Author: > PS. Looking elsewhere, if the program ID is in two places, make it be in one place rather than duplicate the code.

fixed ✅

Reviewer: I can still see "make sure you copy and paste the program ID for your program into the appropriate area of the source code" above - there is no information about what file and what line to modify.

Reviewer: Oh wait I can see it's below. I'll make a suggestion to make this more obvious.

program.

> If you use the frontend, simply replace the `MOVIE_REVIEW_PROGRAM_ID` in both

Reviewer: Having the same constant in two places is lame. Can you please fix that in the https://github.com/solana-developers/movie-review-frontend?

Author: > Having the same constant in two places is lame. Can you please fix that in the https://github.com/solana-developers/movie-review-frontend?

fixed ✅

You can test your program by submitting a transaction with the right instruction
data. For that, feel free to use

> [this script](https://github.com/Unboxed-Software/solana-movie-client) or

Reviewer: This URL needs to be updated to https://github.com/solana-developers/movie-review-program-client.

It's already been partially updated but it does need a format.

Author: okay, I'd work on that

Author: Just created a pull request which fixes the mentioned issues. 

https://github.com/solana-developers/movie-review-program-client/pull/1

Author: > This URL needs to be updated to https://github.com/solana-developers/movie-review-program-client and that repo needs to be updated too. It's already been partially updated but it does need a bunch of other small updates (format, import specific items instead of '*', remove IIFE wrappers, use esrun, etc). - basically what you did already for the frontend repo.

url updated ✅

[the frontend](https://github.com/solana-developers/movie-review-frontend) we
built in the

> [Deserialize Custom Instruction Data lesson](/content/courses/native-onchain-development/deserialize-custom-data-frontend.md).
> In both cases, make sure you copy and paste the program ID for your program into
> the appropriate area of the source code to make sure you're testing the right

Reviewer: ```suggestion
[Deserialize Custom Instruction Data lesson](/content/courses/native-onchain-development/deserialize-custom-data-frontend.md).
In both cases, set the program ID for your program in
the appropriate file:
```

Author: fixed

program.

> If you're using the script, simply replace the value assigned to movieProgramId
> in the index.ts component with the public key of the program you've deployed.
> 
> If you use the frontend, simply replace the `MOVIE_REVIEW_PROGRAM_ID` in the
> `review-form.tsx` components with the address of the program you’ve deployed.

Reviewer: ```suggestion
 - If you're using the script, simply replace the value assigned to `movieProgramId`
in the `index.ts` component with the public key of the program you've deployed.

 - If you use the frontend, simply replace the `MOVIE_REVIEW_PROGRAM_ID` in the
`review-form.tsx` components with the address of the program you’ve deployed.
```

Author: fixed
